### PR TITLE
Adjust noteForm return and fix tests

### PR DESCRIPTION
**Pull request by geet084**
fixed the tests so that they all now pass... 100% 
also moved the maps out of the return to make the return a little easier to look at
-- changed the form so that it removes the blank item onSubmit
-- also changed the form when isEdit is TRUE in app to generate a blank item at the beginning

### CODE REVIEW

File: src/containers/NoteForm/NoteForm.js [hunk 1]
```diff
@@ -77,6 +76,10 @@ export class NoteForm extends Component {
     const { items } = this.state;
     const { isEdit, editNote, postNote } = this.props;
     const url = isEdit ? `http://localhost:3001/api/v1/notes/${id}` : 'http://localhost:3001/api/v1/notes'
+    
+    if (items[items.length - 1].description === '') {
+      items.pop()
+    }
```
Comment (foxwellm) [inline src/containers/NoteForm/NoteForm.js:82]: I think we might be manipulating state directly here

Comment (geet084) [reply src/containers/NoteForm/NoteForm.js:82]: Fixed now sir

**Review by foxwellm — APPROVED** [review]

File: src/containers/NoteForm/NoteForm.test.js [hunk 2]
```diff
@@ -49,21 +47,47 @@ describe('noteForm', () => {
...
+          isCompleted: false,
+          noteID: 1,
+          timestamp: 5
+        }],
+        redirect: false,
+      }
+
+      wrapper.instance().handleTitleChange(mockEvent)
+      expect(wrapper.state()).toEqual(expected)
+    })
```
Comment (ericweissman) [inline src/containers/NoteForm/NoteForm.test.js:90]: Awesome work getting us to 100%!

File: src/containers/NoteForm/NoteForm.js [hunk 3]
```diff
@@ -74,11 +73,15 @@ export class NoteForm extends Component {
...
-    const { items } = this.state;
+    const newItems = this.state.items;
     const { isEdit, editNote, postNote } = this.props;
     const url = isEdit ? `http://localhost:3001/api/v1/notes/${id}` : 'http://localhost:3001/api/v1/notes'
-    isEdit ? editNote(url, { id, title, items }) : postNote(url, { id, title, items })
-    this.setState({ redirect: true })
+    
+    if (newItems[newItems.length - 1].description === '') {
+      newItems.pop()
+    }
```
Comment (ericweissman) [inline src/containers/NoteForm/NoteForm.js:82]: Looks like a good way to handle that last "empty" item on typing into the input.

**Review by ericweissman — APPROVED** [review]

**Pull request merged**